src/SonarQube.Common/MsBuildPathSettings.cs

```diff
@@ -30,8 +30,12 @@ public IEnumerable<string> GetImportBeforePaths()
         {
             var appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
 
-            yield return GetMsBuildImportBeforePath(appData, "15.0");
+            yield return GetMsBuildImportBeforePath(appData, "4.0");
+            yield return GetMsBuildImportBeforePath(appData, "10.0");
+            yield return GetMsBuildImportBeforePath(appData, "11.0");
+            yield return GetMsBuildImportBeforePath(appData, "12.0");
             yield return GetMsBuildImportBeforePath(appData, "14.0");
+            yield return GetMsBuildImportBeforePath(appData, "15.0");
```

Comment:
Not sure if this is the best solution. The check for incompatible MSBuild version already exists in the targets, we just make sure they are executed for all possible versions of MSBuild.